lib/byron/src/Cardano/Wallet/Byron/Compatibility.hs

```diff
@@ -0,0 +1,356 @@
+byronEpochStability =
+    Quantity 2160
+
+-- | Hard-coded active slot coefficient (a.k.a 'f' in Ouroboros/Praos)
+byronActiveSlotCoefficient :: W.ActiveSlotCoefficient
+byronActiveSlotCoefficient =
+    W.ActiveSlotCoefficient 1.0
+
+--------------------------------------------------------------------------------
+--
+-- Genesis
+
+genesisTip :: Tip ByronBlock
+genesisTip = Tip genesisPoint genesisBlockNo
+
+-- FIXME
+-- Actually figure out a way to get this from the network. For Haskell nodes,
+-- there's actually no such thing as a genesis block. But there's a genesis
+-- UTxO and a genesis hash. So, we might be able to ajust our abstractions to
+-- this.
+genesisBlock :: ByronHash -> ByronBlock
+genesisBlock genesisHash = ByronBlock
+    { byronBlockRaw = ABOBBoundary $ ABoundaryBlock
+        { boundaryBlockLength = 0
+        , boundaryHeader = UnsafeABoundaryHeader
+          { boundaryPrevHash = Left (Genesis.GenesisHash (coerce genesisHash))
+          , boundaryEpoch = 0
+          , boundaryDifficulty = ChainDifficulty 0
+          , boundaryHeaderAnnotation = mempty
+          }
+        , boundaryBody = ABoundaryBody mempty
+        , boundaryAnnotation = mempty
+        }
+    , byronBlockSlotNo = genesisSlotNo
+    , byronBlockHash = genesisHash
+    }
+
+--------------------------------------------------------------------------------
+--
+-- Network Parameters
+
+-- | Settings for configuring a MainNet network client
+mainnetVersionData
+    :: (NodeToClientVersionData, CodecCBORTerm Text NodeToClientVersionData)
+mainnetVersionData =
+    ( NodeToClientVersionData { networkMagic = NetworkMagic 764824073 }
+    , nodeToClientCodecCBORTerm
+    )
+
+-- | Settings
+testnetVersionData
+    :: (NodeToClientVersionData, CodecCBORTerm Text NodeToClientVersionData)
+testnetVersionData =
+    ( NodeToClientVersionData { networkMagic = NetworkMagic 1097911063 }
+    , nodeToClientCodecCBORTerm
+    )
+
+--------------------------------------------------------------------------------
+--
+-- Type Conversions
+
+toByronHash :: W.Hash "BlockHeader" -> ByronHash
+toByronHash (W.Hash bytes) =
+    case Crypto.digestFromByteString bytes of
+        Just digest ->
+            ByronHash $ AbstractHash digest
+        Nothing ->
+            error "unsafeHash: failed to convert bytes to hash?"
+
+toEpochSlots :: W.EpochLength -> EpochSlots
+toEpochSlots =
+    EpochSlots . fromIntegral . W.unEpochLength
+
+toPoint :: W.BlockHeader -> Point ByronBlock
+toPoint (W.BlockHeader sid _ h _)
+    | sid == W.SlotId 0 0 = genesisPoint
+    | otherwise = O.Point $ Point.block (toSlotNo sid) (toByronHash h)
+
+toSlotNo :: W.SlotId -> SlotNo
+toSlotNo =
+    SlotNo . W.flatSlot byronEpochLength
+
+fromByronBlock :: W.Hash "Genesis" -> ByronBlock -> W.Block
+fromByronBlock genesisHash byronBlk = case byronBlockRaw byronBlk of
+  ABOBBlock blk  ->
+    mkBlock $ fromTxAux <$> unTxPayload (blockTxPayload blk)
+  ABOBBoundary _ ->
+    mkBlock []
```

Comment:
Does our DB work well with boundary blocks? (Multiple blocks in the same slot I believe)